src/main.py

```diff
@@ -90,8 +90,8 @@ def run(options: Options) -> int:
                 print(f"Function {name} not found.", file=sys.stderr)
                 return 1
 
-    return_code = 0
-    global_info = GlobalInfo(mips_file.asm_data, typemap)
+    function_names = {fn.name for fn in mips_file.functions}
+    global_info = GlobalInfo(mips_file.asm_data, function_names, typemap)
```

Comment:
For this, I'm not totally sure which is better:
I could give the whole `mips_file` to `GlobalInfo` -- but that seems overkill? On the other hand, the python API is getting a bit complex for 3rd parties to set up and use.

Reply:
Both options feel good to me, I don't think it matters much either way.